python/src/main/java/org/apache/zeppelin/python/PythonInterpreter.java

```diff
@@ -54,8 +55,9 @@
 
   private Integer port;
   private GatewayServer gatewayServer;
-  private Boolean py4J = false;
+  private Boolean py4JisInstalled = false;
   private InterpreterContext context;
+  private Pattern errorInLastLine = Pattern.compile(".*(Error|Exception): .*$");
```

Comment:
Maybe we should put user code under `try ... except` or check for non-empty `sys.stderr`?, as an alternative?